### ISSUE TITLE
What is version 1.0.0?

### ISSUE THREAD

**Issue by lynxlynxlynx** [ja2-stracciatella/ja2-stracciatella#443]
As part of the 0.16 milestone / #437 , we have to decide on how to version the next releases.

1.0 carries symbolic value, representing completion, stability.

But what concrete criteria should we meet before being eligible? If you ask me, we're not there yet and the criteria would look something like:
- no external contribution left untouched, meaning some decision is made on what to do with it
  - vxx cherry-picks are done #307 
  - android bits partly #435 
  - there was MP work done, but since we know little about it, nor have much interest, it could easily remain out of scope or something for post 1.0
  - no untriaged pull requests
- no known ja2s induced regressions or crashers
  - from which I'd exclude small graphical quirks from higher resolution support
  - AP related bugs, ours
  - issues with noother n-vanilla tag 
- a crossplatform launcher

The rest is a bonus.

Discuss.

**Comment by selaux** [comment]
I agree with most of the stuff. Not sure if we can fix all non-vanilla regressions though. Imho there will always be some, so we could relax this a bit.

**Comment by selaux** [comment]
BTW: Do you have a link to the MP stuff, I would be interested to at least have a look.

**Comment by lynxlynxlynx** [comment]
Emphasis on "known". Even now we're at only 17, so it looks feasible. But sure, a bridge to cross once we're closer.

I guess it was this:
http://thepit.ja-galaxy-forum.com/index.php?t=thread&frm_id=226

**Comment by JAsmine-ja2** [comment]
### My 5c on "what should Stracciatella v1.0.0 be like":

**Should absolutely have (in descending order):**
- A launcher, which makes messign around with command lines (outside of the launcher itself) obsolete
- As much stuff externalized (#289) as feasible. This would greatly support the "modding-platform" aspect of Stracciatella.
- as few Stracciatella-introduced bugs as feasible. If some remain, then there is something for post-1.0.0-releases to do :)
- support for Android (maybe not on the same level as the other OS's)

**Doesn't need to have:**
- any form of multiplayer (1.13 has it and the last years clearly show that interest in MP is generally low. Apart from that, MP has probably TONS of specific issues to it.)
- every (known) vanilla bug fixed. There will always be vanilla issues remaining. As of now, Stracciatella is already much, much "cleaner" than vanilla, anyway :) Again, more stuff for post-1.0.0-releases :)

**Would ideally also (_but doesn't absolutely need to_) have:**
- UI scaling #62 (maybe in a beta-state - would make high resolution much more useful and would be an outstanding stracciatella-specific feature overall)

**Comment by flaviojs** [comment]
In general, as soon as a piece of software claims it's stable it should immediately move on from 0.X versions.

The definition of stable can be different.
My personal view is that excluding special releases (like alpha, or beta, or a pre release) it should be considered stable as soon as an official release is made.

It appears that we claim to be stable so maybe we should already be >= 1.0?
https://github.com/ja2-stracciatella/ja2-stracciatella/blob/1c98146b1c4910987383800bc128b87a61626773/CMakeLists.txt#L31

An alternative to switching to 1.0 is to move the minor version to the major version. (v0.17 => v17)
I forgot which, but I think a famous piece of software made a move like this before.
